src/configmypy/type_inference.py

```diff
@@ -0,0 +1,22 @@
+# Infer types of argparse arguments intelligently
+
+from argparse import ArgumentTypeError
+
+
+def infer_boolean(var):
+    if var.lower() == 'true':
+        return True
+    elif var.lower() == 'false':
+        return False
+    else:
+        raise ArgumentTypeError()
```

Comment:
So I think one thing is that we'd want to allow e.g. `int` : if a user has False or None in the config, they might want to pass in another param.

Perhaps we could express these as just a "priority" list of types?